src/spi2.rs

```diff
@@ -0,0 +1,681 @@
+//! SPI peripheral control
```

Comment:
I have a general question as to the approach here: why are we creating a completely new implementation of SPI, in its own module? Given that the SPI impl in `master` is not compatible with the one in the `release` branch, I think we should just modify the existing implementation in a way that it has the extra capabilities you suggest. As long as it is still possible to implement the e-hal 1 and e-hal 0.2 traits on top of it, that is. This would also save me the effort of looking at a wall of code, which surely is 80% the same as the one inthe current driver, and trying to figure out where you've done changes?

Reply:
yeah i was planing to exactly does this. to make my life easier  i was just developing in a parallel filer to fetch other potential upstream changes to spi.rs. sorry to confuse you here